modules/serviceregistry/src/main/java/org/opencastproject/serviceregistry/impl/ServiceRegistryJpaImpl.java

```diff
@@ -3419,6 +3458,68 @@ public int compare(ServiceRegistration serviceA, ServiceRegistration serviceB) {
 
   }
 
+  /**
+   * Comparator that will sort service registrations depending on their capacity, which is defined by the number of jobs
+   * the service's host is already running divided by the MaxLoad of the Server. The lower that number, the bigger the capacity.
+   * This Comparator will preferre encoding workers, if none are defined in the configuration file it will act like the LoadComparator.
+   */
+  private static final class LoadComparatorEncoding implements Comparator<ServiceRegistration> {
+
+    private SystemLoad loadByHost = null;
+
+    /**
+     * Creates a new comparator which is using the given map of host names and loads.
+     *
+     * @param loadByHost
+     */
+    LoadComparatorEncoding(SystemLoad loadByHost) {
+      this.loadByHost = loadByHost;
+    }
+
+    @Override
+    public int compare(ServiceRegistration serviceA, ServiceRegistration serviceB) {
+      String hostA = serviceA.getHost();
+      String hostB = serviceB.getHost();
+      NodeLoad nodeA = loadByHost.get(hostA);
+      NodeLoad nodeB = loadByHost.get(hostB);
+
+      if (encodingWorkers != null) {
+        if (isEncodingWorker(hostA, encodingWorkers) && !isEncodingWorker(hostB, encodingWorkers)) {
+          if (nodeA.getLoadFactor() <= encodingThreshold) {
+            return -1;
+          }
+          return Float.compare(nodeA.getLoadFactor(), nodeB.getLoadFactor());
+        }
+        if (isEncodingWorker(hostB, encodingWorkers) && !isEncodingWorker(hostA, encodingWorkers)) {
+          if (nodeB.getLoadFactor() <= encodingThreshold) {
+            return 1;
+          }
+          return Float.compare(nodeA.getLoadFactor(), nodeB.getLoadFactor());
+        }
+      }
+
+      //If the load factors are about the same, sort based on maximum load
```

Comment:
I would be very, very tempted to subclass `LoadComparator`, and then replace this copy+pasted code with a call to `super.compare()` so that there's less chance of comparator implementation drift over time.